files/default/neutron-ha-tool.py

```diff
@@ -1136,32 +1141,34 @@ def _kill_pids_in_namespace(self):
 
             remaining -= 1
             if remaining:
-                pids = self._get_namespace_pids()
+                pids = self._get_namespace_pids(namespace)
                 if pids:
                     LOG.debug("Some processes are still running in namespace "
-                              "%s on host %s. Retrying.", self.namespace,
+                              "%s on host %s. Retrying.", namespace,
                               self.target_host)
                     time.sleep(1)
             else:
                 break
 
-    def delete_router_namespace(self):
-        LOG.debug("Deleting namespace %s on host %s.",
-                  self.namespace,
+    def delete_remote_namespace(self, namespace):
+        LOG.debug("Deleting namespace %s on host %s.", namespace,
                   self.target_host)
-        self.ssh_client = paramiko.SSHClient()
-        self.ssh_client.set_missing_host_key_policy(
-            paramiko.AutoAddPolicy())
-        self.ssh_client.load_system_host_keys()
-        self.ssh_client.connect(self.target_host, timeout=self.timeout)
+        self._ssh_connect()
         with self.ssh_client:
             try:
-                if self._namespace_exists():
-                    self._kill_pids_in_namespace()
-                    self._simple_ssh_command(self.netns_del)
+                if self._namespace_exists(namespace):
+                    self._kill_pids_in_namespace(namespace)
+                    self._simple_ssh_command(self.netns_del + namespace)
             except socket.timeout:
                 LOG.warn("SSH timeout exceeded. Failed to delete namespace "
-                         "%s on %s", self.namespace, self.target_host)
+                         "%s on %s", namespace, self.target_host)
+
+
+class RemoteRouterNsCleanup(RemoteNodeCleanup):
```

Comment:
we could have avoided the interface change by overriding the constructor and setting the router id as a member variable.